File: Python/automotive/qsort/qsort_small.py
import sys

MAX_ARRAY = 60000


class MyStringClass:
    def __init__(self, qstring):
        self.qstring = qstring

def main():
    if len(sys.argv) < 2:
        print("Usage: python qsort_small.py <file>")
        sys.exit(-1)

    filename = sys.argv[1]
    array = []

    with open(filename, "r") as fp:
        count = 0
        for line in fp:
            if count >= MAX_ARRAY:
                break
            array.append(MyStringClass(line.strip()))
            count += 1

    print("\nSorting {} elements.\n".format(count))
    array.sort(key=lambda x: x.qstring, reverse=True)

    for item in array:
        print(item.qstring)


if __name__ == "__main__":
    main()

File: Java/automotive/qsort/src/QSortSmall.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;

public class QSortSmall {
    private static final int MAX_ARRAY = 60000;

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java QSortSmall <file>");
            System.exit(-1);
        }

        String filename = args[0];
        String[] array = new String[MAX_ARRAY];
        int count = 0;

        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = reader.readLine()) != null && count < MAX_ARRAY) {
                array[count] = line.trim();
                count++;
            }
        } catch (IOException e) {
            e.printStackTrace();
            System.exit(-1);
        }

        System.out.println("\nSorting " + count + " elements.\n");

        String[] filledArray = Arrays.copyOfRange(array, 0, count);

        Arrays.sort(filledArray, (a, b) -> b.compareTo(a));

        for (String str : filledArray) {
            System.out.println(str);
        }
    }
}
